src/reporter.ts

```diff
@@ -69,50 +69,65 @@ function scrub(payload: any, headers: any, sizeLimit: number, truncated = false)
-      if (
-        shouldScrubDomain(host) ||
-        (request?.host && shouldScrubDomain(request.host)) ||
-        (response?.host && shouldScrubDomain(response.host))
-      ) {
-        request.body = 'The data is not available';
-        response.body = 'The data is not available';
-        delete request.headers;
-        delete response.headers;
-        delete request.uri;
-      } else {
-        const isError = spanHasErrors(span);
-        const sizeLimit = getEventEntitySize(isError);
-        if (span.info?.httpInfo?.response?.body) {
-          span.info.httpInfo.response.body = scrub(
-            decodeHttpBody(response.body, isError),
-            response.headers,
-            sizeLimit,
-            span.info.httpInfo.response.truncated
-          );
-        }
-        if (span.info?.httpInfo?.request?.body) {
-          span.info.httpInfo.request.body = scrub(
-            decodeHttpBody(request.body, isError),
-            request.headers,
-            sizeLimit
-          );
-        }
+const scrubSpan = (span) => {
+  if (span.info?.httpInfo) {
+    const { request, response, host } = span.info.httpInfo;
+    if (
+      (response && request && shouldScrubDomain(host)) ||
+      (request?.host && shouldScrubDomain(request.host)) ||
+      (response?.host && shouldScrubDomain(response.host))
+    ) {
+      request.body = 'The data is not available';
+      response.body = 'The data is not available';
+      delete request.headers;
+      delete response.headers;
+      delete request.uri;
+    } else {
+      const isError = spanHasErrors(span);
+      const sizeLimit = getEventEntitySize(isError);
+      if (span.info.httpInfo.response?.body) {
+        span.info.httpInfo.response.body = scrub(
+          decodeHttpBody(response.body, isError),
+          response.headers,
+          sizeLimit,
+          span.info.httpInfo.response.truncated
+        );
+      }
+      if (span.info.httpInfo.request?.body) {
+        span.info.httpInfo.request.body = scrub(
+          decodeHttpBody(request.body, isError),
+          request.headers,
+          sizeLimit
+        );
+      }
+      if (span.info.httpInfo.request?.headers) {
         span.info.httpInfo.request.headers = payloadStringify(request.headers, sizeLimit);
-        if (response?.headers)
-          span.info.httpInfo.response.headers = payloadStringify(response.headers, sizeLimit);
       }
+      if (response?.headers)
+        span.info.httpInfo.response.headers = payloadStringify(response.headers, sizeLimit);
+    }
+  }
+  return span;
+};
+
+// We muted the spans itself to keep the memory footprint of the tracer to a minimum
+export function scrubSpans(resultSpans: any[]) {
+  let from = 0,
+    to = 0;
+  while (from < resultSpans.length) {
+    if (safeExecute(scrubSpan, 'Failed to scrub span')(resultSpans[from])) {
+      resultSpans[to] = resultSpans[from];
+      to++;
     }
-  });
+    from++;
+  }
+  resultSpans.length = to;
```

Comment:
use plain filters - this is after the trimming, I.e. max 1MB memory. This code seems dangerous.